src/cc-index.ts

```diff
@@ -55,44 +57,227 @@ export class CCIndex {
             }
 
 
-        }, ccIndexPageURL);
+        }, timeout, ccIndexPageURL);
 
     }
 
 
     /**
+     * Takes an array of URL to look up, queries the CC index, parses the CC responses and returns
+     * an array of relative paths to WET files that should include the URLs. This function handles all errors.
+     *
+     * @param lookupURLs                    URLs to look up in the CC index
+     * @param takeOnlyTheFirstWetPath       If set to TRUE, only one WET path will be returned for each URL (there might be multiple WETs for one URL)
+     * @param callback                      Will be called with an array of relative wet paths (might be empty)
+     * @param timeout               (optional) max waiting time for the response
+     * @param ccIndexPageURL                (optional) base URL of the CC index page; if not provided -> defaultCCIndex is used
+     */
+    public static getWETPathsForEachURL(lookupURLs : Array<string>,
+                                        takeOnlyTheFirstWetPath : boolean,
+                                        callback : (wetPaths : Array<string>) => void,
+                                        timeout? : number,
+                                        ccIndexPageURL? : string) {
+
+        let wets : Set<string> = new Set<string>();
+        console.log("start looking up " + lookupURLs.length + " urls");
+
+        // We can't send 100500 requests at the same time, instead we:
+        //  - query cc index with one url,
+        //  - wait for the response,
```

Comment:
Why though? Wouldn't it be much more efficient to have, let's say, 50 connections open at a time?